### DISCUSSION TITLE
View/Edit properties Search Results

### DISCUSSION THREAD

**Discussion by NareshLSR** [microsoft-search/pnp-modern-search#1515] [Q&A] [answered]
I recently downloaded the latest version of pnp-modern-search webpart (V4) and it is awesome.

I am using the search results webpart, with a details list layout to show documents. I am looking for options to view/edit the properties of the document, download, print from an ECB menu (link to item). Is this doable in the latest version. If yes, please help me with some tips.

**Comment by wobba** [comment, marked as answer]
Doable if you develop custom web components to support this and use them in your template. So...a fair amount of work involved.

**Reply by hsamdani12** [reply]
Hi, a follow up on this. We are looking for same functionality of ECB Menu (in OOTB Modern Document Library Views- when user clicks on "Details", they get the document preview and the ability to view/edit all the metadata detail ? Is there plan to build this into the search result webpart ?  If not, is this doable using handlebar customization/configurarion or would it need SPFx development ?

**Reply by wobba** [reply]
You would need to create a web component with SPFx to achieve this, and no current plans.

**Reply by hsamdani12** [reply]
@wobba thanks!
